baseapp-chats/baseapp_chats/graphql/mutations.py

```diff
@@ -283,7 +294,54 @@ def read_messages(cls, room, profile, message_ids=None):
         return ChatRoomReadMessages(room=room, profile=profile, messages=messages)
 
 
+class ChatRoomUnread(RelayMutation):
+    room = graphene.Field(ChatRoomObjectType)
+    profile = graphene.Field(ProfileObjectType)
+
+    class Input:
+        room_id = graphene.ID(required=True)
+        profile_id = graphene.ID(required=True)
+
+    @classmethod
+    @login_required
+    def mutate_and_get_payload(cls, root, info, room_id, profile_id, **input):
+        room = get_obj_from_relay_id(info, room_id)
+        profile = get_obj_from_relay_id(info, profile_id)
+
+        if not info.context.user.has_perm("baseapp_profiles.use_profile", profile):
```

Comment:
@nossila should we find a way to reuse those checks? it's being used in multiple places right now